sim/src/main.rs

```diff
@@ -896,9 +1189,25 @@ pub struct ImageHeader {
     _pad3: u32,
 }
 
-impl AsRaw for ImageHeader {}
+impl AsRaw for SmallImageHeader {}
+
+#[repr(C)]
+pub struct LargeImageHeader {
+    small_header: SmallImageHeader,
+    _pad0: u32,
```

Comment:
`_pad: [u32; 8],` would be a little cleaner, when writing the values, use `[0; 8]` for the value of all zeros.